packages/x-date-pickers/src/DateCalendar/useCalendarState.tsx

```diff
@@ -71,18 +72,27 @@ export const createCalendarStateReducer =
           !disableSwitchToMonthOnDayFocus &&
           !utils.isSameMonth(state.currentMonth, action.focusedDay);
 
+        let slideDirection: SlideDirection;
+        if (!needMonthSwitch) {
+          slideDirection = state.slideDirection;
```

Comment:
Before, the slide to the new month was caused by the `useEffect` tracking the external value.
So the focus of the clicked day was already done
But now it is the click itself  which caused the slide (like when you click on the array left / right).

Without this check, the focus was resetting the slide direction, causing it to go on the wrong direction in some scenario.